src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -310,26 +311,21 @@ FancyZones::WindowCreated(HWND window) noexcept
 {
     if (m_settings->GetSettings().appLastZone_moveWindows && IsInterestingWindow(window))
     {
-        auto monitor = MonitorFromWindow(window, MONITOR_DEFAULTTONULL);
-        if (monitor)
+        for (const auto& [monitor, zoneWindow] : m_zoneWindowMap)
         {
-            auto zoneWindow = m_zoneWindowMap.find(monitor);
-            if (zoneWindow != m_zoneWindowMap.end())
+            const auto activeZoneSet = zoneWindow->ActiveZoneSet();

```

Comment:
Are we safe here with not checking if zoneWindow is not null?

Reply:
Well, we're not adding it into `m_zoneWindowMap` if it's `nullptr` (check `AddZoneWindow` method).